### ISSUE TITLE
[MNG-1171] I wish there was a recommended way of naming plugins

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#3278]
**[Brian Bonner](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bkbonner)** opened **[MNG-1171](https://issues.apache.org/jira/browse/MNG-1171?redirect=false)** and commented

What **is** the preferred article id naming for Maven Plugins?

I've seen it as \<pluginname>-maven-plugin   and maven-\<pluginname>-plugin
It would help me (and other users) avoid silly mistakes if there was a recommended pattern.

I ran into this with the xmlbeans plugin.  The example had it named as maven-xmlbeans-plugin.  After looking at other plugins, it looks like this is the "standard".

It seems like the name should either go at the beginning or end  (i.e. xmlbeans-maven-plugin,  or maven-plugin-xmlbeans).

I'm wondering if there is really a rhyme or reason to this.  It would make it easier to write poms if there was some sort of recommended standard.

---
No further details from [MNG-1171](https://issues.apache.org/jira/browse/MNG-1171?redirect=false)

**Comment by jira-importer** [comment]
**[Jason van Zyl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jvanzyl)** commented

The standard naming convention is maven-\<foo>-plugin and we've been using it for a long time.

**Comment by jira-importer** [comment]
**[Vincent Massol](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=vmassol)** commented

Hi Jason,

AFAIK the naming convention for external plugins is **-maven-plugin and not maven-**-plugin which is reserved for the maven-provided plugins. At least that has always been my understanding...

**Comment by jira-importer** [comment]
**[Brian Bonner](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bkbonner)** commented

FWIW, When I opened this up, I had only been working with Maven2 for a little while and Maven2 was still in Beta3.  Since then, I've observed what Vincent described.

When I was writing this, it wasn't clear (or documented) why they were named this way.

When filling out pom.xml, as a maven user, I guess I thought the artifactId naming pattern would be consistent and the group ID (org.apache.maven.plugins) would be enough to indicate that it was a maven-provided-plugin.

Thanks.
